### ISSUE TITLE
Verbosity is used inconsistently in the CLI

### ISSUE THREAD

**Issue by KathleenDollard** [dotnet/cli#9165]
-v is used inconsistently between CLI and bundled tools and is not consistent across CLI commands. 

This makes an inconstant user experience and confusing help. Since we anticipate global tool authors following these patterns, it sets us up for an unnecessary fractured ecosystem.

### CLI 

The CLI usage is:

```bash
 -v|--verbosity        Set the verbosity level of the command. Allowed values are q[uiet], m[inimal], n[ormal], d[etailed], and diag[nostic].
```

The CLI produces an error if a verbosity level not given: this is especially ugly because we print help and obscure the error.

At least `dotnet new` and `dotnet buidserver` do not support --vervbosity.
 
### Bundled tools: watch, sql-cache, user-secrets 

The usage in `dotnet watch`, `dotnet ef`, `dotnet dev-certs`, `dotnet sql-cache`, `dotnet user-secrets`is:
 
-v|--verbose     Show verbose output.

## Proposal:

Bundled tools and CLI support --verbosity consistently with no argument being "[d]etailed", and that this be the guidance for command line creation for Global Tools. As a result, these will all be equivalent
 
--verbosity
--verbosity d
-v
-v d
--verbose (bundled tools only, and possibly marked obsolete)
 
This requires the CLI support a default value for the --verbosity switch of "d"
 
This requires bundled tool to support --verbosity with a -v in addition to --verbose.

### Special handling

To avoid backward compatibility issues with `dotnet run` and `dotnet install`, it's important that the parser be greedy. For example, in this case, "detailed" should not be passed on to the application being run:

```bash
dotnet run -v detailed
dotnet install -v detailed
```

Assuming errors no longer occurring are not considered breaking changes, then this is not a breaking change. The following would have previously resulted in an error, and would now pass "banana" on to the application:

```bash
dotnet run -v banana
```

Passing no verbosity level also results in an error today:

```bash
dotnet run -v
```
 
### Breaking changes

From the perspective of bundled tools, especially `dotnet watch` a breaking change may occur if the verbosity level passed is also an executable that would be run. The following code would attempt to run an application named "detailed" today and would set the detailed switch under this proposal.

```bash
dotnet watch -v detailed
```

Please weigh in if you see any more breaking change scenarios.

**Comment by kasper3** [comment]
The FTUE is broken in help itself. First time user tries `dotnet -h`, and finds: 

```
Common options:
  -v|--verbosity        Set the verbosity level of the command. Allowed values are q[uiet], m[inimal], n[ormal], d[etailed], and diag[nostic].
  -h|--help ....
```

`--help`, with shorthand `-h`, is some option that applies to every command and `dotnet.exe` itself; so being that under `Common option` of `dotnet --help` makes sense.

`--verbosity` on the other hand is not a common option. `dotnet` itself, `dotnet new` etc. do not provide `--verbosity` option. Seeing `--vebosity` in top level `dotnet --help` is confusion-prone and incorrect, but putting that under the heading "Common options" of `dotnet --help` (which itself doesn't support --verbosity) is really unexpected and a strange choice.

**Comment by peterhuene** [comment]
I agree with @kasper3.  I think `--verbosity` should be limited to commands that forward to MSBuild.  Commands that do not, such as `new` and `build-server` shouldn't support the options since it won't affect their output.  I think we should make it clear that in the context of the `dotnet` commands that support it, it is really controlling the MSBuild verbosity.

Personally I'd delete it from the top-level help since it really is a command-specific option.

Regarding the bundled tools: I'd like some consistency there, but having them support the MSBuild verbosity options is probably overkill for their design.

**Comment by KathleenDollard** [comment]
@peterhuene That sounds great. I look forward to seeing what you come up with.

**Comment by peterhuene** [comment]
So I propose something like this:

```
.NET Command Line Tools (2.1.400-preview-008657)
Usage: dotnet [sdk-options] [command] [command-options] [arguments]

Runs a .NET Core SDK command.

sdk-options:
  -h|--help         Show command line help.
  --version         Display .NET Core SDK version in use.
  --info            Display .NET Core information.
  --list-sdks       Display the installed SDKs.
  --list-runtimes   Display the installed runtimes.
  -d|--diagnostics  Enable diagnostic output.

SDK commands:
  new               Initialize .NET projects.
  restore           Restore dependencies specified in a .NET project.
  run               Compile and run a .NET project output.
  build             Build a .NET project.
  publish           Publish a .NET project for deployment.
  test              Run unit tests using the test runner specified in a .NET project.
  pack              Create a NuGet package.
  migrate           Migrate a project.json project to an MSBuild project.
  clean             Clean build outputs of a .NET project.
  sln               Modify Visual Studio solution files.
  add               Add a package or reference to a .NET project.
  remove            Remove a package or reference from a .NET project.
  list              List project references of a .NET project.
  nuget             Provides additional NuGet commands.
  msbuild           Run Microsoft Build Engine (MSBuild) commands.
  vstest            Run Microsoft Test Engine (VSTest) commands.
  store             Store the specified assemblies in the runtime package store.
  tool              Install or manage tools that extend the .NET experience.
  build-server      Interact with servers started by a build.
  help              Show command line help.

Run 'dotnet [command] --help' for more information on a command.

Usage: dotnet [runtime-options] [path-to-application] [arguments]

Runs a .NET Core application.

runtime-options:
  --additionalprobingpath <path>     Path containing probing policy and assemblies to probe for.
  --fx-version <version>             Version of the installed Shared Framework to use to run the application.
  --roll-forward-on-no-candidate-fx  Roll forward on no candidate shared framework is enabled.
  --additional-deps <path>           Path to additional deps.json file.

path-to-application:
  The path to an application .dll file to execute.
```

Changed the following:

* The usage is now separated into two different sections: one for using SDK commands and one for running .NET applications.  The existing help output is a jumble of the two (which comes first I have no preference on).
* Fix all of the command description strings to be consistent with verb tense and nomenclature.
* Better align the whitespace.
* Add a description to each of the two supported usages.
* Removes the "common options" section since it would only show `--help` because `--verbosity` is not a common option (see above comment).  Since we have a `Run 'dotnet [command] --help' for more information on a command.` blurb, it's a bit redundant to have the "common options" section, so I just moved it to the `sdk-options`.

Thoughts?

I'd also love to sort the SDK command names (alphabetically maybe?).  Right now it's in "wherever the implementor inserted it" order.

**Comment by KathleenDollard** [comment]
Let's target this for 2.1.400.

I'd also like to consider:

* Alphabetizing the commands now that there are so many
* Including the bundled tools
* Having a notion of "not-common" that is orthogonal to bundling
* Having "not-common" commands and the runtime options be under a "dotnet --help --verbosity detailed" which we can hopefully replace with "dotnet --help -v" as a shortcut and having this suggestion at the bottom of help

**Comment by peterhuene** [comment]
This will be targetted for 2.1.400.  The bundled tools will be included, just waiting for the merge up of Will's changes that went into 2.1.3xx.

Regarding the `--verbosity` option, it's not currently a top-level option.  `dotnet` also has no notion of logging levels (except for "verbose", which is enabled with `-d` for debug output).  I don't think we should be adding this into `dotnet` because it's really `MSBuild` specific.

**Comment by drewlsvern** [comment]
In regards to this being used in consistently, verbosity doesn't seem to currently work in SDK version 2.1.105 with the command dotnet run.

The following command below will start the web app just fine, but it doesn't provide any console output once the app is running.

Ex. dotnet run --verbosity detailed or dotnet run --verbosity diagnostic

I see mixed documentation everywhere. Can anyone confirm if it should work?

**Comment by peterhuene** [comment]
Hi @drewlsvern.  The `--verbosity` option was added to the `run` command in the upcoming `2.1.300` release of the .NET Core SDK.  The top level help does seem to imply that the option is universally supported, but that's actually not the case; that's what we hope to address in the help for 2.1.400.

Regarding your web application, the `--verbosity` option just controls the output of the implicit restore and build the `run` command performs (i.e. the MSBuild output).  It will not affect any output being done by the target executable.  Thus, if your web application is writing to stdout, I'd expect it to show up with `dotnet run` in 2.1.105 and later.  If that's not the case, would you mind opening an issue with repro steps and I'll investigate.  Thanks!

**Comment by drewlsvern** [comment]
Seems an uninstall and reinstall of the sdk fixed the issue.
Thanks.
